hazelcast/src/test/java/com/hazelcast/jet/pipeline/test/TestSourcesTest.java

```diff
@@ -156,17 +159,27 @@ public void test_itemStream_withWindowing() throws Throwable {
     public void test_itemStream_in_expected_range() throws Throwable {
         int itemsPerSecond = 109;
 
+        int windowSize = 1000;
         p.readFrom(TestSources.itemStream(itemsPerSecond))
-                .withNativeTimestamps(0)
-                .window(tumbling(1000))
-                .aggregate(counting())
-                .apply(assertCollectedEventually(10, windowResults -> {
+         .withNativeTimestamps(0)
+         .window(tumbling(windowSize))
+         .aggregate(counting())
+         .apply(assertCollectedEventually(10, windowResults -> {
                     // first window may be incomplete
                     assertTrue("sink list should contain some items", windowResults.size() > 1);
-                    assertTrue("emitted items is more than twice lower then expected itemsPerSecond",
-                            (long) windowResults.get(1).result() > itemsPerSecond / 2);
-                    assertTrue("emitted items is more than twice higher then expected itemsPerSecond",
-                            (long) windowResults.get(1).result() < itemsPerSecond * 2);
+                    WindowResult<Long> windowResult = windowResults.get(1);
+
+                    long pauses = JitterRule.getPausesBetween(windowResult.start(), windowResult.end());
+                    // Ignore if a long pause was recorded
```

Comment:
you could use `assumeThat` so the test is marked as ignored instead silently completed. This would need support in `AssertionSinks`

Reply:
Right, but the resulting effect would be the same - we don't track ignored tests in any way.